edb/edgeql/compiler/context.py

```diff
@@ -111,7 +111,7 @@ def __call__(
 
 class PendingCardinality(NamedTuple):
 
-    specified_cardinality: Optional[qltypes.Cardinality]
+    specified_cardinality: Optional[qltypes.SchemaCardinality]
```

Comment:
This place, in particular, should use `qltypes.Cardinality` instead

Reply:
Hmm, the `specified_cardinality` is taken from the schema in the current implementation. That's why I left it as `SchemaCardinality`. 

However, if we wanted to convert the schema cardinality to inference cardinality, that's where we'd need those helpers dealing with bounds and also later on to check if the inferred cardinality falls within the bounds of the specified one. Technically, we could get away with just the `is_single`/`is_multi` helpers, but only if we don't allow `required` as part of the spec. If I can specify that my computable is a `required multi` (it was my understanding that this is actually one of the end goals of this cardinality change in the first place), then I need proper bound checking to ensure that inferred `AT_MOST_ONE` is not acceptable as specified `AT_LEAST_ONE`, but `ONE` is. Incidentally, this would make CardinalityBound again move closer to Cardinality in terms of packaging so that the utilities to compare cardinalities can use the bounds conveniently.

Reply:
I don't see why `inference.cardinality` could not export a helper function to compare cardinalities.  `qltypes` is supposed to be a simple set of dumb enums to be reused across schema and various ASTs.

Reply:
OK, I've renamed everything back to "cardinality".